src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = {};
+
+  sourceString.split(';').forEach((item) => {
+    const pairs = item.trim().split(':');
+    let key;
+    let value;
+
+    if (pairs.length === 2) {
+      key = pairs[0].trim();
+      value = pairs[1].trim();
+    }
+    obj[key] = value;
```

Comment:
The current implementation incorrectly assigns `undefined` to `obj[key]` when a string without a colon is encountered due to the scope of `key` and `value`. Ensure that you only assign key-value pairs to the object when both a key and a value are present.